### DISCUSSION TITLE
Help needed: Live Search after 3 input characters

### DISCUSSION THREAD

**Discussion by manishlok** [livewire/livewire#1985] [1. Help] [answered]
From a newbie;
I've implemented a live search input and getting result as expected.

`<input type="text" class="form-input" placeholder="Search Users..." wire:model="query"
        wire:keydown.escape="resetValues" wire:keydown.tab="resetValues" wire:keydown.arrow-up="decrementHighlight"
        wire:keydown.arrow-down="incrementHighlight" wire:keydown.enter="selectUser" />`

my question is:
How to get search result only if input exceeds 3 characters so that network request can be reduced ?

**Comment by Ahrengot** [comment, marked as answer]
You need a client side solution for this. You could do it with alpine. 

Here's what I would do:
1. Set up alpine with an entangled query that syncs the search query between alpine and livewire. `x-data={query: @entangle('query')}`
2. listen for change on the input with `x-on:change` and implement your own debouncer, so requests aren't fired on every keystroke after 3 characters, but only after the user stops typing
3. When the debouncer fires, call the search function from alpine
